Insufficiently accurate exception message in case of IOException

**I want to report a bug.**
If a file cannot be found, the IllegalStateException is thrown. But the message contains `Unable to read file org.sonar.javascript.visitors.JavaScriptFileImpl@5a2bd7c8` instead of filename. 

You can find from stacktrace, which file cause the error, but the description of IllegalStateException is insufficiently accurate.

**SonarJS version:** 6.2.1
**SonarQube version:** 8.2<br>

**Reproducer** 
Run a check for a project with a file, that causes IOException. For me, it was an inaccessible symbolic link with `.js` extension.
```
// Directory listing
test.js -> test2.js // test2.js does not exists
```

**Logs**
```
ERROR: Error during SonarQube Scanner execution
java.lang.IllegalStateException: Unable to read file org.sonar.javascript.visitors.JavaScriptFileImpl@5a2bd7c8
	at org.sonar.javascript.checks.utils.CheckUtils.readLines(CheckUtils.java:138)
	at org.sonar.plugins.javascript.minify.AverageLineLengthCalculator.getAverageLineLength(AverageLineLengthCalculator.java:53)
	at org.sonar.plugins.javascript.minify.MinificationAssessor.hasExcessiveAverageLineLength(MinificationAssessor.java:66)
	at org.sonar.plugins.javascript.minify.MinificationAssessor.isMinified(MinificationAssessor.java:53)
	at org.sonar.plugins.javascript.JavaScriptExclusionsFileFilter.accept(JavaScriptExclusionsFileFilter.java:47)
	at org.sonar.scanner.scan.filesystem.FileIndexer.accept(FileIndexer.java:258)
	at org.sonar.scanner.scan.filesystem.FileIndexer.indexFile(FileIndexer.java:138)
	at org.sonar.scanner.scan.filesystem.ProjectFileIndexer$IndexFileVisitor.visitFile(ProjectFileIndexer.java:242)
	at org.sonar.scanner.scan.filesystem.ProjectFileIndexer$IndexFileVisitor.visitFile(ProjectFileIndexer.java:214)
	at java.base/java.nio.file.Files.walkFileTree(Unknown Source)
	at org.sonar.scanner.scan.filesystem.ProjectFileIndexer.indexDirectory(ProjectFileIndexer.java:210)
	at org.sonar.scanner.scan.filesystem.ProjectFileIndexer.indexFiles(ProjectFileIndexer.java:197)
	at org.sonar.scanner.scan.filesystem.ProjectFileIndexer.index(ProjectFileIndexer.java:159)
	at org.sonar.scanner.scan.filesystem.ProjectFileIndexer.indexModulesRecursively(ProjectFileIndexer.java:138)
	at org.sonar.scanner.scan.filesystem.ProjectFileIndexer.index(ProjectFileIndexer.java:108)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:346)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
	at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:141)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
	at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:72)
	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:66)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:189)
	at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:138)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:112)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:75)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
Caused by: java.nio.file.NoSuchFileException: /Users/tarexme/TestProject/test.js
	at java.base/sun.nio.fs.UnixException.translateToIOException(Unknown Source)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(Unknown Source)
	at java.base/java.nio.file.Files.newByteChannel(Unknown Source)
	at java.base/java.nio.file.Files.newByteChannel(Unknown Source)
	at java.base/java.nio.file.spi.FileSystemProvider.newInputStream(Unknown Source)
	at java.base/java.nio.file.Files.newInputStream(Unknown Source)
	at org.sonar.api.batch.fs.internal.DefaultInputFile.inputStream(DefaultInputFile.java:98)
	at org.sonar.api.batch.fs.internal.DefaultInputFile.contents(DefaultInputFile.java:108)
	at org.sonar.javascript.visitors.JavaScriptFileImpl.contents(JavaScriptFileImpl.java:42)
	at org.sonar.javascript.checks.utils.CheckUtils.newBufferedReader(CheckUtils.java:143)
	at org.sonar.javascript.checks.utils.CheckUtils.readLines(CheckUtils.java:134)
	... 34 more
```

**Expected behavior**
The message of IllegalStateException should contain the file path. For example: `java.lang.IllegalStateException: Unable to read file test.js`

--- a/javascript-checks/pom.xml
+++ b/javascript-checks/pom.xml
@@ -56,6 +56,10 @@
       <artifactId>commons-io</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.mockito</groupId>
+      <artifactId>mockito-core</artifactId>
+    </dependency>
   </dependencies>
 
 </project>

--- a/javascript-checks/src/main/java/org/sonar/javascript/checks/utils/CheckUtils.java
+++ b/javascript-checks/src/main/java/org/sonar/javascript/checks/utils/CheckUtils.java
@@ -135,7 +135,7 @@ public class CheckUtils {
       return reader.lines().collect(Collectors.toList());
 
     } catch (IOException e) {
-      throw new IllegalStateException("Unable to read file " + file.toString(), e);
+      throw new IllegalStateException("Unable to read file " + file.uri(), e);
     }
   }
 

--- a/javascript-checks/src/test/java/org/sonar/javascript/checks/utils/CheckUtilsTest.java
+++ b/javascript-checks/src/test/java/org/sonar/javascript/checks/utils/CheckUtilsTest.java
@@ -22,12 +22,14 @@ package org.sonar.javascript.checks.utils;
 import com.sonar.sslr.api.typed.ActionParser;
 import java.io.File;
 import java.io.IOException;
+import java.net.URI;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.util.Arrays;
 import java.util.List;
 import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 import org.junit.rules.TemporaryFolder;
 import org.sonar.api.batch.fs.internal.DefaultInputFile;
 import org.sonar.javascript.checks.verifier.TestUtils;
@@ -36,8 +38,11 @@ import org.sonar.javascript.visitors.JavaScriptFileImpl;
 import org.sonar.plugins.javascript.api.tree.ScriptTree;
 import org.sonar.plugins.javascript.api.tree.Tree;
 import org.sonar.plugins.javascript.api.tree.declaration.FunctionDeclarationTree;
+import org.sonar.plugins.javascript.api.visitors.JavaScriptFile;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 import static org.sonar.javascript.checks.utils.CheckUtils.readLines;
 
 public class CheckUtilsTest {
@@ -46,6 +51,9 @@ public class CheckUtilsTest {
 
   @Rule
   public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Rule
+  public ExpectedException exceptionRule = ExpectedException.none();
 
   @Test
   public void testAsString() throws Exception {
@@ -73,5 +81,16 @@ public class CheckUtilsTest {
 
     DefaultInputFile testInputFile = TestUtils.createTestInputFile(temporaryFolder.getRoot(), tmp.getName(), StandardCharsets.UTF_8);
     assertThat(readLines(new JavaScriptFileImpl(testInputFile))).isEqualTo(lines);
+  }
+
+  @Test
+  public void testReadLinesThrows() throws IOException {
+    exceptionRule.expect(IllegalStateException.class);
+    exceptionRule.expectMessage("Unable to read file fileUri");
+
+    JavaScriptFile file = mock(JavaScriptFile.class);
+    when(file.uri()).thenReturn(URI.create("fileUri"));
+    when(file.contents()).thenThrow(IOException.class);
+    readLines(file);
   }
 }

